CMakeLists.txt

```diff
@@ -167,7 +167,14 @@ endif()
 find_package(PCRE)
 
 include(FindOpenSSL)
+include(CheckOpenSSLIsBoringSSL)
 find_package(OpenSSL)
+check_openssl_is_boringssl(OPENSSL_IS_BORINGSSL "${OPENSSL_INCLUDE_DIR}")
+
+if(OPENSSL_VERSION VERSION_GREATER_EQUAL "3.0.0")
+    set(OPENSSL_IS_OPENSSL3 ON)
+    add_compile_definitions(OPENSSL_API_COMPAT=10002 OPENSSL_IS_OPENSSL3)
```

Comment:
`./plugins/certifier/CMakeLists.txt:add_definitions(-DOPENSSL_API_COMPAT=10002)`

Interesting, this is duplicated in the certifier. That's not necessarily bad because it helps decouple, but I'm wondering whether the one in certifier should also be guarded by a conditional.

Reply:
This is how auto tools is setup when openssl3 is found.  I'm not sure if its necessary.